packages/nimble-components/src/table/models/update-tracker.ts

```diff
@@ -0,0 +1,133 @@
+import { DOM } from '@microsoft/fast-element';
+import type { Table } from '..';
+import type { TableColumn } from '../../table-column/base';
+import type { TableRecord } from '../types';
+
+interface BooleanCollection {
+    [key: string]: boolean;
+}
+
+interface RequiredUpdates extends BooleanCollection {
+    rowIds: boolean;
+    columnIds: boolean;
+    columnSort: boolean;
+    columnDefinition: boolean;
+    actionMenuSlots: boolean;
+}
+
+/**
+ * Helper class to track what updates are needed to the table based on configuration
+ * changes.
+ */
+export class UpdateTracker<TData extends TableRecord> {
+    private readonly requiredUpdates: RequiredUpdates = {
+        rowIds: false,
+        columnIds: false,
+        columnSort: false,
+        columnDefinition: false,
+        actionMenuSlots: false
+    };
+
+    private readonly table: Table<TData>;
+    private updateQueued = false;
+
+    public constructor(table: Table<TData>) {
+        this.table = table;
+    }
+
+    public get updateRowIds(): boolean {
+        return this.requiredUpdates.rowIds;
+    }
+
+    public get updateColumnIds(): boolean {
+        return this.requiredUpdates.columnIds;
+    }
+
+    public get updateColumnSort(): boolean {
+        return this.requiredUpdates.columnSort;
+    }
+
+    public get updateColumnDefinition(): boolean {
+        return this.requiredUpdates.columnDefinition;
+    }
+
+    public get updateActionMenuSlots(): boolean {
+        return this.requiredUpdates.actionMenuSlots;
+    }
+
+    public get requiresTanStackUpdate(): boolean {
+        return (
+            this.requiredUpdates.rowIds
+            || this.requiredUpdates.columnSort
+            || this.requiredUpdates.columnDefinition
+        );
+    }
+
+    public get requiresTanStackDataReset(): boolean {
+        return (
+            this.requiredUpdates.rowIds || this.requiredUpdates.columnDefinition
+        );
+    }
+
+    public connectedCallback(): void {
+        this.setAllKeys(true);
```

Comment:
Would it be better to just create an `updateAll` method that combines the logic in `setAllKeys` (which on its own isn't the most clear in its purpose by name) and a call to `queueUpdate`?

Reply:
To make sure I'm understanding, you're suggesting renaming `connectedCallback` here to `updateAll`. Is that correct? If so, I think that suggestion makes a lot of sense.

At some point I also had a `disconnectedCallback()` as well, so I decided to use the same naming convention as we had done for the virtualizer utility. But, I don't think that naming makes as much sense anymore.